internal/dot/graph.go

```diff
@@ -60,19 +81,29 @@ type Group struct {
 	Type    reflect.Type
 	Name    string
 	Results []*Result
+	State   FailureState
 }
 
 // Graph is the DOT-format graph in a Container.
 type Graph struct {
-	Ctors  []*Ctor
-	Groups map[groupKey]*Group
+	Ctors   []*Ctor
+	ctorMap map[CtorID]*Ctor
+
+	Groups   []*Group
+	groupMap map[groupKey]*Group
+
+	Failed *failedNodes
 }
 
-// Node is a single node in a graph and is embedded into Params and Results.
-type Node struct {
-	Type  reflect.Type
-	Name  string
-	Group string
+type failedNodes struct {
```

Comment:
It looks like all of the failures are of type `Result` - should they be of type `Node`? If `Result` is more desirable, could you explain? I'm fine with `Result`, but want to make sure it's necessary.

Reply:
As above, `GroupIndex` in `Result` is needed to identify which grouped node failed.

Reply:
Cool, makes sense.